### PR TITLE
Add utility functions for reading testing env vars

### PR DESCRIPTION
**Pull request by aznashwan**
Depends on https://github.com/canonical/k8s-workflows/pull/15

### CODE REVIEW

File: k8s_test_harness/util/docker_util.py
Comment (bschimke95) [inline k8s_test_harness/util/docker_util.py:17]: nit: you can use https://github.com/canonical/k8s-test-harness/blob/d503366f0b481183567de806175ac8d6fee547dc/k8s_test_harness/util/exec_util.py#L25 for this.

Comment (bschimke95) [reply k8s_test_harness/util/docker_util.py:17]: and below.

File: k8s_test_harness/util/env_util.py
Comment (bschimke95) [inline k8s_test_harness/util/env_util.py:41]: Just a reminder comment to update before merging this

File: k8s_test_harness/util/env_util.py
Comment (bschimke95) [inline k8s_test_harness/util/env_util.py:90]: nit: you can use `typing.Self` here.

File: k8s_test_harness/util/env_util.py
Comment (bschimke95) [inline k8s_test_harness/util/env_util.py]: nit: extra `;`

File: k8s_test_harness/util/env_util.py
Comment (berkayoz) [inline k8s_test_harness/util/env_util.py]:
```suggestion
        return os.path.join(self.rock_dir, "rockcraft.yaml")
```
Isn't this supposed to be `rockcraft.yaml`?

**Review by berkayoz — COMMENTED** [review]
Looks good overall, left a comment

File: k8s_test_harness/util/env_util.py
Comment (bschimke95) [inline k8s_test_harness/util/env_util.py:164]: just a comment: this basically rebuilds the whole `RockMetaInfo` list from the environment every time you call this function. Maybe returning a `RockMetaInfos` data object would make sense and this object has a `get_info_for_rock_version`. 

Anyway, this is not performance critical code - just leave it like this.

**Review by bschimke95 — COMMENTED** [review]
Minor nits (and a comment) then we are good to merge (after the workflow is merged).

Great work!

File: k8s_test_harness/util/env_util.py
Comment (aznashwan) [reply k8s_test_harness/util/env_util.py:90]: That was [what I did initially](https://github.com/canonical/k8s-test-harness/pull/5/commits/9e524854e5927536b80231ee02100951d0efce99) but the GH runner is using Python 3.8 where it does not exist and I had to switch to using a plain string for the code to actually run. (in terms of type hinting tooling the effect should be the same)

File: k8s_test_harness/util/env_util.py
Comment (aznashwan) [reply k8s_test_harness/util/env_util.py:41]: This requires a tag/release on the workflows (which is why the URL is currently linking to the `main` branch).

Without it, I could either link the current HEAD of the repo or remove the comment entirely.

File: k8s_test_harness/util/env_util.py
Comment (aznashwan) [reply k8s_test_harness/util/env_util.py]: Great catch, brain fart, updated the function name for clarity as well!

File: k8s_test_harness/util/docker_util.py
Comment (aznashwan) [reply k8s_test_harness/util/docker_util.py:17]: Sadly there's something weird with how pytest loads modules going on and it [leads to partially initalized import issues](https://github.com/aznashwan/harbor-rocks/actions/runs/9989824882/job/27609690604#step:7:53).

In the interest of time, I have converted it back to using `subprocess` directly. (the added logging that `exec_util.run()` offers won't be printed on successful tests anyway, and the whole command line argument list will be reported by `subprocess` on errors anyway, so there's no real notable disadvantage without it anyway)

File: k8s_test_harness/util/env_util.py
Comment (aznashwan) [reply k8s_test_harness/util/env_util.py:164]: Agreed, but most sanity tests will only call this once anyways, and I can add it later if I feel the integration tests could really use it.

**Review by bschimke95 — APPROVED** [review]
LGTM

**Pull request merged**
